akka-typed/src/main/scala/akka/typed/javadsl/BehaviorBuilder.scala

```diff
@@ -0,0 +1,141 @@
+   *
+   * @param `type` type of messages to match
+   * @param predicate a predicate that will be evaluated on the argument if the type matches
+   * @param handler action to apply if the type matches and the predicate returns true
+   * @tparam M
+   * @return a new behavior with the specified handling appended
+   */
+  def message[M <: T](`type`: Class[M], predicate: FI.TypedPredicate[M], handler: FI.Apply2[ActorContext[T], M, Behavior[T]]): BehaviorBuilder[T] =
+    copy[T](messageHandlers = messageHandlers :+ Case[T, T](`type`, Some((t: T) ⇒ predicate.defined(t.asInstanceOf[M])), (i1: ActorContext[T], msg: T) ⇒ handler.apply(i1, msg.asInstanceOf[M])))
+
+  /**
+   * Add a new case to the message handling without compile time type check.
+   * Should normally not be used, but when matching on class with generic type
+   * argument it can be useful, e.g. <code>List.class</code> and <code>(List&lt;String&gt; list) -> {...}</code>
+   *
+   * @param `type` type of messages to match
+   * @param handler action to apply when the type matches
+   * @return a new behavior with the specified handling appended
+   */
+  def messageUnchecked[M <: T](`type`: Class[_ <: T], handler: FI.Apply2[ActorContext[T], M, Behavior[T]]): BehaviorBuilder[T] =
+    copy[T](messageHandlers = messageHandlers :+ Case[T, T](`type`, None, (i1: ActorContext[T], msg: T) ⇒ handler.apply(i1, msg.asInstanceOf[M])))
+
+  /**
+   * Add a new case to the message handling matching equal messages.
+   *
+   * @param msg the message to compare to
+   * @param handler action to apply when the message matches
+   * @return a new behavior with the specified handling appended
+   */
+  def messageEquals(msg: T, handler: FI.Apply[ActorContext[T], Behavior[T]]): BehaviorBuilder[T] =
+    copy[T](messageHandlers = messageHandlers :+ Case[T, T](msg.getClass, Some(_.equals(msg)), (ctx: ActorContext[T], _: T) ⇒ handler.apply(ctx)))
+
+  /**
+   * Add a new case to the signal handling
+   *
+   * @param `type` type of signal to match
+   * @param handler action to apply if the type matches
+   * @tparam M
+   * @return a new behavior with the specified handling appended
+   */
+  def signal[M <: Signal](`type`: Class[M], handler: FI.Apply2[ActorContext[T], M, Behavior[T]]): BehaviorBuilder[T] =
+    copy[T](signalHandlers = signalHandlers :+ Case[T, Signal](`type`, None, (ctx: ActorContext[T], signal: Signal) ⇒ handler.apply(ctx, signal.asInstanceOf[M])))
+
+  /**
+   * Add a new predicated case to the signal handling.
+   *
+   * @param `type` type of signals to match
+   * @param predicate a predicate that will be evaluated on the argument if the type matches
+   * @param handler action to apply if the type matches and the predicate returns true
+   * @tparam M
+   * @return a new behavior with the specified handling appended
+   */
+  def signal[M <: Signal](`type`: Class[M], predicate: FI.TypedPredicate[M], handler: FI.Apply2[ActorContext[T], M, Behavior[T]]): BehaviorBuilder[T] =
```

Comment:
This should be `akka.japi.function.Predicate` and call the param `test` perhaps?